services/core-web/src/components/mine/Permit/PermitConditionManagement.js

```diff
@@ -0,0 +1,281 @@
+import React, { Component } from "react";
+import { bindActionCreators } from "redux";
+import { connect } from "react-redux";
+import PropTypes from "prop-types";
+import { Divider, Collapse, Button, Row, Col } from "antd";
+import { flattenObject } from "@common/utils/helpers";
+import { ReadOutlined } from "@ant-design/icons";
+import { openModal, closeModal } from "@common/actions/modalActions";
+import {
+  getPermitConditionCategoryOptions,
+  getPermitConditionTypeOptions,
+} from "@common/selectors/staticContentSelectors";
+import { getPermitConditions, getEditingConditionFlag } from "@common/selectors/permitSelectors";
+import {
+  fetchPermitConditions,
+  deletePermitCondition,
+  updatePermitCondition,
+  setEditingConditionFlag,
+  getPermitAmendment,
+} from "@common/actionCreators/permitActionCreator";
+import {
+  fetchMineRecordById,
+} from "@common/actionCreators/mineActionCreator";
+import { maxBy } from "lodash";
+import AddCondition from "@/components/Forms/permits/conditions/AddCondition";
+import ConditionLayerOne from "@/components/Forms/permits/conditions/ConditionLayerOne";
+import CustomPropTypes from "@/customPropTypes";
+import { modalConfig } from "@/components/modalContent/config";
+import { COLOR } from "@/constants/styles";
+import { Link } from "react-router-dom";
+import { ArrowLeftOutlined } from "@ant-design/icons";
+import * as route from "@/constants/routes";
+import { formatDate } from "@common/utils/helpers";
+
+const propTypes = {
+  openModal: PropTypes.func.isRequired,
+  closeModal: PropTypes.func.isRequired,
+  conditions: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.any)).isRequired,
+  permitConditionCategoryOptions: PropTypes.arrayOf(CustomPropTypes.dropdownListItem).isRequired,
+  editingConditionFlag: PropTypes.bool.isRequired,
+  fetchPermitConditions: PropTypes.func.isRequired,
+  setEditingConditionFlag: PropTypes.func.isRequired,
+  deletePermitCondition: PropTypes.func.isRequired,
+  updatePermitCondition: PropTypes.func.isRequired,
+  hasSourceConditions: PropTypes.bool.isRequired,
+  getPermitAmendment: PropTypes.func.isRequired,
+  fetchMineRecordById: PropTypes.func.isRequired,
+};
+
+export class PermitConditionManagement extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { 
+      submitting: false, 
+      permitNo: '', 
+      issuesDate: '', 
+      authEndDate: '',
+      mineName: ''
+    };
+    this.fetchPermitConditions();
+    props.setEditingConditionFlag(false);
+  }
+  
+  componentDidMount = () =>{
+    this.props.fetchMineRecordById(this.props.match.params.mine_guid).then((response) => {
+      this.setState({ 
```

Comment:
Can you set this in a state variable? and reference it form the state throughout the file? same with 'this.props.match.params.id' 

```this.setState({ 
        permitAmendmentGuid: this.props.match.params.id,
        mineGuid: this.props.match.params.mine_guid,
      });
      ```